### ISSUE TITLE
CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#9124]
```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds , count left: 1
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:812)
    at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:805)
    at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:797)
    at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:104)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast/1069/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK8/com.hazelcast$hazelcast/963/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:111)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-ZuluJDK6/com.hazelcast$hazelcast/373/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:111)
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/659/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:111)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.8/com.hazelcast$hazelcast/404/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds , count left: 1
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:820)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:813)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:805)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:104)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/682/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:111)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.7/com.hazelcast$hazelcast/1120/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds , count left: 1
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:820)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:813)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:805)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:104)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast/1134/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:111)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.8/com.hazelcast$hazelcast/461/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by tkountis** [comment]
Functionality seems to be correct, I don't see any race condition that could cause this. System seems to be going in overdrive, big delays in operations, big cluster time shifts and many migrations from which many aren't successful due to Nodes being still in STARTING mode. At some point, partitions are lost completely, which leads to the wrong figures. This is the same behaviour with both #9017 and #9217. 

I checked system load on the diagnostic logs of the last build, and load was ok, no reason to have caused these delays. I don't know what caused this behaviour, but it also seems related to #9202. I would close the issue.

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds, count left: 2
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:923)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:916)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:908)
	at com.hazelcast.cardinality.CardinalityEstimatorAdvancedTest.testCardinalityEstimatorSpawnNodeInParallel(CardinalityEstimatorAdvancedTest.java:107)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast/1167/testReport/junit/com.hazelcast.cardinality/CardinalityEstimatorAdvancedTest/testCardinalityEstimatorSpawnNodeInParallel/

**Comment by tombujok** [comment]
Fixed via #9661 by @jerrinot
